docs/how-to-guides.md

```diff
@@ -0,0 +1,383 @@
+to your library, although wrapped libraries are heavily encouraged to avoid
+global namespace pollution.
+
+#### `sources`
+
+Dune works slightly differently than ReScript when it comes down to including
+source folders as part of a library.
+
+By default, when Dune finds a `dune` file with a `library` stanza, it will
+include just the files inside that folder to the library itself (unless the
+`modules` field is used). If you want to create a library with multiple
+subfolders in it, you can use the following combination of stanzas:
+
+- `(include_subdirs unqualified)`
+  ([docs](https://dune.readthedocs.io/en/stable/dune-files.html#include-subdirs)):
+  This stanza tells Dune to look for sources in all the subfolders of the folder
+  where the `dune` file lives.
+- `(dirs foo bar)`
+  ([docs](https://dune.readthedocs.io/en/stable/dune-files.html#dirs)): This
+  stanza tells Dune to only look into `foo` and `bar` subdirectories of the
+  current folder.
+
+So for example, if your library had this configuration in its `bsconfig.json`:
+
+```json
+{
+  "sources": ["src", "lib"]
+}
+```
+
+You might translate this to a `dune` file with the following configuration:
+
+```text
+(include_subdirs unqualified)
+(dirs src lib)
+(library
+ (name lib)
+ (modes melange))
+```
+
+Alternatively, depending on the case, you could place two separate `dune` files,
+one in `src` and one in `lib`, and define one `library` on each. In that case,
+`include_subdirs` and `dirs` would not be necessary.
+
+Regarding the `"type" : "dev"` configuration in ReScript, the way Dune solves
+that is with public and private libraries. If a `library` stanza includes a
+`public_name` field, it becomes a public library, and will be installed.
+Otherwise it is private and won't be visible by consumers of the package.
+
+#### `bs-dependencies`
+
+Your library might depend on other libraries. To specify dependencies of the
+library in the `dune` file, you can use the `libraries` field of the `library`
+stanza.
+
+For example, if `bsconfig.json` had something like this:
+
+```json
+"bs-dependencies": [
+  "reason-react"
+]
+```
+
+Your `dune` file will look something like:
+
+```text
+(library
+ (name lib)
+ (libraries reason-react)
+ (modes melange))
+```
+
+Remember that you will have to add all dependencies to your library `opam`
+package as well.
+
+#### `bs-dev-dependencies`
+
+There is no such thing as `bs-dev-dependencies` in Dune. If you need to bring
+some libraries or packages for development, you can just `opam install` them, or
+use the [upcoming `--with-dev-setup`
+configuration](https://opam.ocaml.org/doc/Manual.html#pkgvar-with-dev-setup) in
+opam 2.2. For example, to add `melange-jest` as dev dependency, you could add
+this in your library `opam` file:
+
+```text
+depends: [
+  "melange-jest" {with-dev-setup}
+]
+```
+
+Then, the library `melange-jest` would become available to your library tests.
```

Comment:
I suppose this is a bit of a weird example because we'd most likely want `{with-test}` in this case.

Reply:
Hm right great point. I will change this to `with-test` instead (which is what `bs-dev-deps` is mainly used for).

Reply:
https://github.com/melange-re/melange-re.github.io/pull/111/commits/274f1b660349ad9bfa95397ab01bcb984b3a31c0